lib/phoenix_ecto/plug.ex

```diff
@@ -23,3 +23,40 @@ unless Ecto.SubQueryError in excluded_exceptions do
     end
   end
 end
+
+unless Phoenix.Ecto.PendingMigrationError in excluded_exceptions do
+  defimpl Plug.Exception, for: Phoenix.Ecto.PendingMigrationError do
+    def status(_error), do: 404
+
+    def actions(%{repo: repo}),
+      do: [
+        %{
+          label: "Run migrations for repo",
+          handler: {__MODULE__, :migrate, [repo]}
+        }
+      ]
+
+    def migrate(repo) do
+      if Process.whereis(repo),
+        do: Ecto.Migrator.run(repo, :up, all: true, dynamic_repo: repo.get_dynamic_repo())
+    end
+  end
+end
+
+unless Phoenix.Ecto.StorageNotCreatedError in excluded_exceptions do
+  defimpl Plug.Exception, for: Phoenix.Ecto.StorageNotCreatedError do
+    def status(_error), do: 404
```

Comment:
I don't think it should be a 404, rather 500:

```suggestion
    def status(_error), do: 500
```

Reply:
My rationale was "there is a missing database here, `404` seems to fit" but I will change it :smile:

Reply:
[fixed here](https://github.com/phoenixframework/phoenix_ecto/compare/846278a787c3300abfadce9a9d8097da9131afe2..5083d99f7f515143f5d00420d620f5491833597f)